Source/Samples/17_SceneReplication/SceneReplication.cpp

```diff
@@ -444,30 +448,87 @@ void SceneReplication::MoveCamera()
+    textElement->SetVerticalAlignment(VA_CENTER);
+    textElement->SetPosition(x, y += yOffset);
+
+    return textElement;
+}
+
+void SetOverlayText(SharedPtr<Text>& textElement, const ea::string& title, int value)
+{
+    textElement->SetText(Format("{}: {}", title, value));
+}
+
+void SceneReplication::UpdateOverlay(int packetsIn, int packetsOut, int bytesIn, int bytesOut, int connections)
+{
+    SetOverlayText(packetsIn_, PACKETS_IN_TITLE, packetsIn);
+    SetOverlayText(packetsOut_, PACKETS_OUT_TITLE, packetsOut);
+    SetOverlayText(bytesIn_, BYTES_IN_TITLE, bytesIn);
+    SetOverlayText(bytesOut_, BYTES_OUT_TITLE, bytesOut);
+    SetOverlayText(connections_, CONNECTIONS_TITLE, connections);
+    const auto* network = GetSubsystem<Network>();
+    serverRunning_->SetText(network->IsServerRunning() ? "Server on" : "");
+}
+
+void SampleConnection(const Connection* connection, int& packetsIn, int& packetsOut, int& bytesIn, int& bytesOut)
+{
+    packetsIn += connection->GetPacketsInPerSec();
+    packetsOut += connection->GetPacketsOutPerSec();
+    bytesIn += connection->GetBytesInPerSec();
+    bytesOut += connection->GetBytesOutPerSec();
+}
+
 void SceneReplication::HandlePostUpdate(StringHash eventType, VariantMap& eventData)
 {
     // We only rotate the camera according to mouse movement since last frame, so do not need the time step
     MoveCamera();
 
-    if (packetCounterTimer_.GetMSec(false) > 1000 && GetSubsystem<Network>()->GetServerConnection())
+    if (packetCounterTimer_.GetMSec(false) < 1000)
+    {
+        return;
+    }
+
+    packetCounterTimer_.Reset();
+
+    const auto* network = GetSubsystem<Network>();
+
+    int packetsIn = 0;
+    int packetsOut = 0;
+    int bytesIn = 0;
+    int bytesOut = 0;
+    int connectionCount = 0;
+
+    if (auto* connectionToServer = network->GetServerConnection())
     {
-        packetsIn_->SetText("Packets  in: " + ea::to_string(GetSubsystem<Network>()->GetServerConnection()->GetPacketsInPerSec()));
-        packetsOut_->SetText("Packets out: " + ea::to_string(GetSubsystem<Network>()->GetServerConnection()->GetPacketsOutPerSec()));
-        packetCounterTimer_.Reset();
+        SampleConnection(connectionToServer, packetsIn, packetsOut, bytesIn, bytesOut);
+        connectionCount = 1;
     }
-    if (packetCounterTimer_.GetMSec(false) > 1000 && GetSubsystem<Network>()->GetClientConnections().size())
+    else
     {
-        int packetsIn = 0;
-        int packetsOut = 0;
-        auto connections = GetSubsystem<Network>()->GetClientConnections();
-        for (auto it = connections.begin(); it != connections.end(); ++it ) {
-            packetsIn += (*it)->GetPacketsInPerSec();
-            packetsOut += (*it)->GetPacketsOutPerSec();
+        const auto connectionsToClients = network->GetClientConnections();
+        if (!connectionsToClients.empty())
+        {
+            connectionCount = connectionsToClients.size();
+            for (int i = 0; i < connectionCount; ++i)
```

Comment:
The common pattern in the engine is to use "unsigned" for counters and sizes.

Reply:
My guess is that byte and packet counters are unsigned too. Please check if they are. I know it's a nit picking, I hope it won't discourage you. Thank you for contributing!

Reply:
No worries, fixed the style issues

Reply:
Since index is not really necessary C++ range loops could be used as well. This is not a change request though, merely something to be aware of. :)